### PR TITLE
Add `LazyList.span`

### PR DESCRIPTION
**Pull request by jaschdoc**
Resolves #2033 

I'm not sure there is any way to make this effect polymorphic, since the recursive structure of `(span(f, l), ENil)` or something to that effect would nest tuples within in tuples. Thoughts?

### CODE REVIEW

**Comment by jaschdoc** [comment]
@magnus-madsen ready to merge

File: main/src/library/LazyList.flix [hunk 1]
```diff
@@ -955,4 +955,29 @@ namespace LazyList {
         case LList(xs)    => partitionAcc(f, force xs, k)
     }
 
+    ///
+    /// `l1` is the longest prefix of `l` that satisfies the predicate `f`.
+    /// `l2` is the remainder of `l`.
+    ///
+    pub def span(f: a -> Bool & ef, l: LazyList[a]): (LazyList[a], LazyList[a]) & ef =
```
Comment (magnus-madsen) [inline main/src/library/LazyList.flix:962]: Does this force the entire list? I would think perhaps I could be lazy in the second list?

Comment (magnus-madsen) [reply main/src/library/LazyList.flix:962]: Second, if `f` is pure, could this be even more lazy or is that not really practical?

**Comment by magnus-madsen** [comment]
I will merge this, but would like to hear your comments!

File: main/src/library/LazyList.flix (hunk 1 above)
Comment (jaschdoc) [reply main/src/library/LazyList.flix:962]: Oh right, good point. I'll have a look at it and get back to you. Marking this PR as a draft for now.

Comment (jaschdoc) [reply main/src/library/LazyList.flix:962]: I guess you could have something to to the effect of returning `(LList(lazy takeWhile(f, l)), LList(lazy dropWhile(f, l)))`

**Comment by jaschdoc** [comment]
I suggest we merge this and add optimize `span` as part of #2307. There are some other functions like `findRight` and `map` which could all be optimized further.

**Comment by magnus-madsen** [comment]
Merged!

**Pull request merged**